jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/ProjectClassLoadersTest.java

```diff
@@ -0,0 +1,109 @@
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void testIsClassInCompileClasspathWhenTrue() throws  Exception{
+        //Given
+        boolean all = true;
+        File applicationProp =  new File(getClass().getResource("/util/spring-boot-application.properties").getPath());
+        File targetFolder = temporaryFolder.newFolder("target");
+        File classesInTarget = new File(targetFolder, "classes");
+        File applicationPropertiesInsideTarget = new File(classesInTarget, "application.properties");
+        FileUtils.copyFile(applicationProp, applicationPropertiesInsideTarget);
+        URLClassLoader compileclassLoader = ClassUtil.createClassLoader(Arrays.asList(classesInTarget.getAbsolutePath(), applicationProp.getAbsolutePath()), classesInTarget.getAbsolutePath());
+        ProjectClassLoaders object = new ProjectClassLoaders(compileclassLoader);
+
+        //When
+        boolean result  =  object.isClassInCompileClasspath(all);
+
+        //Then
+        assertTrue(result);
+    }
+    @Test
+    public void testIsClassInCompileClasspathWhenFalse() throws  Exception{
+        //Given
+        boolean all = false;
+        File applicationProp =  new File(getClass().getResource("/util/spring-boot-application.properties").getPath());
+        File targetFolder = temporaryFolder.newFolder("target");
+        File classesInTarget = new File(targetFolder, "classes");
+        File applicationPropertiesInsideTarget = new File(classesInTarget, "application.properties");
+        FileUtils.copyFile(applicationProp, applicationPropertiesInsideTarget);
+        URLClassLoader compileclassLoader = ClassUtil.createClassLoader(Arrays.asList(classesInTarget.getAbsolutePath(), applicationProp.getAbsolutePath()), classesInTarget.getAbsolutePath());
+        ProjectClassLoaders object = new ProjectClassLoaders(compileclassLoader);
+
+        //When
+        boolean result  =  object.isClassInCompileClasspath(all,"ProjectClassLoadersTest","UserConfigurationCompare");
+
+        //Then
+        assertFalse(result);
+    }
+
+    @Test
+    public void testIsClassInCompileClasspathWhenHasAllClassesTrue() throws  Exception{
+        //Given
+        boolean all = true;
+        File applicationProp =  new File(getClass().getResource("/util/spring-boot-application.properties").getPath());
+        File targetFolder = temporaryFolder.newFolder("target");
+        File classesInTarget = new File(targetFolder, "classes");
+        File applicationPropertiesInsideTarget = new File(classesInTarget, "application.properties");
+        FileUtils.copyFile(applicationProp, applicationPropertiesInsideTarget);
+        URLClassLoader compileclassLoader = ClassUtil.createClassLoader(Arrays.asList(classesInTarget.getAbsolutePath(), applicationProp.getAbsolutePath()), classesInTarget.getAbsolutePath());
+        ProjectClassLoaders object = new ProjectClassLoaders(compileclassLoader);
```

Comment:
If I'm not mistaken all the tests seem to be having this block where you set up `ProjectClassLoader` maybe you can move this to some function or maybe move it to `@Before`, you can make `ProjectClassLoaders` as a member variable and then initialize it in `@Before` block, it would be called automatically before each test execution